### ISSUE TITLE
Examine new PyTorch launch utility for distributed jobs

### ISSUE THREAD

**Issue by jose5918** [kubeflow/pytorch-operator#30]
http://pytorch.org/docs/stable/distributed.html#launch-utility

At a glance, it doesn't seem to offer a better alternative to using the current initializations methods, but probably requires at least a closer look.

**Comment by jlewi** [comment]
It looks like the tool is spawning multiple processes on nodes. How is it doing that? ssh? We are spawning processess using the K8s APIs. My guess is that the functionality provided by that tool is a poor substitute for the K8s APIs.

**Comment by jlewi** [comment]
Code is [here](https://github.com/pytorch/pytorch/tree/master/torch/distributed)

**Comment by jlewi** [comment]
Based on the comments in [launch.py](https://github.com/pytorch/pytorch/blob/master/torch/distributed/launch.py) it looks like launch.py needs to be manually started on each node. I think the value add of launch.py is that can manage multiple processes on a single node; all it does is call POpen.

So it could in principle be used in conjection with the PyTorch operator.

However, an alternative approach would be to create pods with multiple containers 1 per process in order to run N processes per node. We could also potentially just have multiple pods.  The advantage of both these approaches is that the controller and K8s have more visibility into the individual processes.

/cc @teng-li @andfoy

**Comment by johnugeorge** [comment]
Distributed training is achieved by setting up `MASTER_PORT`, `RANK`, `WORLD_SIZE`, `MASTER_PORT` environment variables while using `init_process_group`.   
As this feature is no more required, closing the issue
/close
